pkg/osquery/runtime/runtime.go

```diff
@@ -831,7 +833,24 @@ func (r *Runner) launchOsqueryInstance() error {
 			case <-o.doneCtx.Done():
 				return o.doneCtx.Err()
 			case <-ticker.C:
-				if err := o.Healthy(); err != nil {
+				// Health check! Allow a couple
+				// failures before we tear everything
+				// down. This is pretty simple, it
+				// hardcodes the timing. Might be
+				// better for a Limiter
+				failed := false
+				for i := 0; i <= 5; i++ {
+					if err := o.Healthy(); err != nil {
+						level.Info(o.logger).Log("msg", "Health check failed", "attempt", i, "err", err)
+						failed = true
+
+					}
+
+					// err was nil, clear failed
+					failed = false
```

Comment:
I think this is going to be called in all cases, think you need
```suggestion
if err == nil {
  failed = false
}
					
```

Reply:
or maybe ...
```suggestion
maxHealthChecks := 5
for i := 1; i <= maxHealthChecks; i++ {
	if err := o.Healthy(); err != nil {
		level.Info(o.logger).Log("msg", "Health check failed", "attempt", i, "err", err)
		
		if i == maxHealthChecks {
			return errors.Wrap(err, "health check failed")
		}
		
	} else {
		break
	}
}
```

Reply:
You're right. I knew I wasn't getting that correct. How's this?

Reply:
Adding in the sleep is ugly. I like your `maxHealthChecks`